swap_meet/item.py

```diff
@@ -1,2 +1,38 @@
+import uuid
+
 class Item:
-    pass
\ No newline at end of file
+    def __init__(self, id=None, condition=0, category=""):
+        self.condition = condition
+        self.category = category
+
+        # if no given id, generate uuid
+        if id is None:
+            self.id = uuid.uuid4().int
+        else:
+            self.id = id
+
+
+    def get_category(self):
+        # return class name
+        # return self.__class__.__name__
+        return "Item"
+
+    
+    def condition_description(self):
+        if self.condition <= 2:
+            return "pre-loved"
+        if 3 <= self.condition <= 4:
+            return "good condition"
+        if self.condition >= 5:
+            return "like new!"
+
+    def __str__(self):
+        # stringify an instance of Item using str()
+        # calling str() on a class instance will cause:
+        # overriding, known as "operator overloading"
+        # the same method exhibits different behavior across instances of different classes
+        id = self.id
+        item = self.__class__.__name__
+
+        result = f"An object of type {item} with id {id}."
+        return result
```

Comment:
No need to create variables when we can directly access an instances's attributes. Also notice that line 35 does the same thing as `get_category()`. Avoid repetition here by invoking the method here instead of rewriting the logic.

```suggestion

        return f"An object of type {self.get_category()} with id {self.id}."
```

With this refactor, you can invoke this method in your child classes to get the first part of the message for the child class's `__str__` method.